packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/webrtc/helpers/BitrateLimiter.kt

```diff
@@ -0,0 +1,65 @@
+package com.fishjamcloud.client.webrtc.helpers
+
+import com.fishjamcloud.client.models.TrackBandwidthLimit
+import org.webrtc.RtpParameters
+import kotlin.math.pow
+
+class BitrateLimiter {
+  companion object {
+    fun calculateBitrates(
+      encodings: List<RtpParameters.Encoding>,
+      maxBitrate: TrackBandwidthLimit
+    ): List<RtpParameters.Encoding> =
+      when (maxBitrate) {
+        is TrackBandwidthLimit.BandwidthLimit ->
+          calculateUniformBitrates(encodings, maxBitrate.limit)
+        is TrackBandwidthLimit.SimulcastBandwidthLimit ->
+          calculateSimulcastBitrates(encodings, maxBitrate.limit)
+      }
+
+    private fun calculateSimulcastBitrates(
+      encodings: List<RtpParameters.Encoding>,
+      limits: Map<String, TrackBandwidthLimit.BandwidthLimit>
+    ): List<RtpParameters.Encoding> =
+      encodings.map { encoding ->
+        val encodingLimit = limits[encoding.rid]?.limit ?: 0
+        encoding.withBitrateInBps(encodingLimit)
+      }
+
+    private fun calculateUniformBitrates(
+      encodings: List<RtpParameters.Encoding>,
+      bitrate: Int
+    ): List<RtpParameters.Encoding> {
+      if (encodings.isEmpty()) return emptyList()
+      if (bitrate == 0) {
+        return encodings.map { it.withBitrateInBps(null) }
+      }
+
+      val k0 = encodings.minByOrNull { it.scaleResolutionDownBy ?: 1.0 }
```

Comment:
I know it is moved code, but maybe we can use some better names here, instead of `k0` and `x`? 🤔

Reply:
I thought about this, but I didn't understand what they mean 🙈 
But I'll try to give it another shot.

Reply:
Thanks for improving it :)